UnitTests/Tests/GetAllProductsQueryHandlerTest.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Moq;
+using NUnit.Framework;
+using SportsGoods.App.Queries;
+using SportsGoods.App.QueryHandlers;
+using SportsGoods.Core.Interfaces;
+using SportsGoods.Core.Models;
+using SportsGoods.Data.DAL;
+using SportsGoods.Data.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SportsGoods.App.Tests.Tests
+{
+    [TestFixture]
+    public class GetAllProductsQueryHandlerTest
+    {
+        private ApplicationDbContext _context;
+        private GetAllProductsQueryHandler _handler;
+
+        [SetUp]
+        public void Setup()
+        {
+            var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+                .UseInMemoryDatabase(databaseName: "TestDatabase")
+                .Options;
+            _context = new ApplicationDbContext(options);
+
+            SeedTestData();
+
+            var repository = new ProductRepository(_context);
+            _handler = new GetAllProductsQueryHandler(repository);
+        }
+
+        [Test]
+        public async Task Handle_ReturnsPagedResultWithProductDTOs()
+        {
+
+            var pageNumber = 1;
+            byte pageSize = 10;
+
+            var mockRepository = new Mock<IProductRepository>();
+            mockRepository.Setup(repo => repo.GetPagedAsync(pageNumber, pageSize))
+                          .ReturnsAsync(GetTestProducts());
+
+            var queryHandler = new GetAllProductsQueryHandler(mockRepository.Object);
+            var query = new GetAllProductsQuery { PageNumber = pageNumber, PageSize = pageSize };
+
+
+            var result = await queryHandler.Handle(query, default);
+
+            Assert.That(result, Is.Not.Null);
+            Assert.That(result.Items.Count, Is.EqualTo(2));
+        }
+
+        [Test]
+        public async Task Handle_ReturnsCorrectPageBasedOnPageSize()
+        {
+            var pageNumber = 1;
+            byte[] pageSizes = { 10, 5, 20 };
```

Comment:
you can use something like 
https://docs.nunit.org/articles/nunit/technical-notes/usage/Parameterized-Tests.html

and 
https://dzone.com/articles/nunit-tutorial-parameterized-tests-with-examples

to make the test more readable

Reply:
Done